pipelines/card_schema.py

```
@@ -0,0 +1,656 @@
+    """Render the visible placeholder for a dataset with no declaration yet."""
+    return "\n".join(
+        [
+            "## Dataset viewer schema",
+            "",
+            "**Not declared yet.** This card carries no `configs` / `dataset_info` "
+            "block, so index availability is unverified here: the datasets-server "
+            "may infer a working schema for a homogeneous payload, while a later "
+            "heterogeneous shard can still make `viewer`, `search`, `filter`, or "
+            "`statistics` fail even where the streaming preview works. The published "
+            "raw payload described above is complete and unmodified; only the "
+            "card-side schema declaration is missing.",
```

Ruff (0.16.2)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)